paasta_tools/autoscaling/autoscaling_service_lib.py

```diff
@@ -353,10 +355,26 @@ def autoscale_marathon_instance(marathon_service_config, marathon_tasks, mesos_t
         **autoscaling_params
     )
 
-    new_instance_count = marathon_service_config.limit_instance_count(current_instances + autoscaling_amount)
+    # Limit downscaling by 30% of current_instances until we find out what is
+    # going on in such situations
+    new_instance_count = max(current_instances + autoscaling_amount, int(current_instances * 0.7))
+
+    new_instance_count = marathon_service_config.limit_instance_count(new_instance_count)
     if new_instance_count != current_instances:
         write_to_log(
```

Comment:
This is going to be pretty noisy. Can you only log this debug if new_instance_count == int(current_instances * 0.7) ?